src/Tmds.Systemd/Journal.cs

```diff
@@ -149,52 +149,60 @@ public static unsafe LogResult Log(LogFlags flags, JournalMessage message)
-                bool loop;
-                do
+                for (int i = 0; i < data.Count; i++)
                 {
-                    loop = false;
-                    msghdr msg;
-                    msg.msg_iov = pIovs;
-                    msg.msg_iovlen = (SizeT)dataLength;
-                    int rv = sendmsg(socket.Handle.ToInt32(), &msg, sendmsgFlags).ToInt32();
-                    if (rv < 0)
+                    handles[i] = GCHandle.Alloc(data[i].Array, GCHandleType.Pinned);
+                    handlesAllocated++;
+                    iovs[i].Base = handles[i].AddrOfPinnedObject();
+                    iovs[i].Length = new IntPtr(data[i].Count);
+                }
+                int sendmsgFlags = 0;
+                if ((flags & LogFlags.DropWhenBusy) != 0)
+                {
+                    sendmsgFlags |= MSG_DONTWAIT;
+                }
+                LogResult result = LogResult.Success;
+                fixed (IOVector* pIovs = &MemoryMarshal.GetReference(iovs))
+                {
+                    bool loop;
+                    do
                     {
-                        int errno = Marshal.GetLastWin32Error();
-                        if (errno == EINTR)
+                        loop = false;
+                        msghdr msg;
+                        msg.msg_iov = pIovs;
+                        msg.msg_iovlen = (SizeT)dataLength;
+                        int rv = sendmsg(socket.Handle.ToInt32(), &msg, sendmsgFlags).ToInt32();
+                        if (rv < 0)
                         {
-                            loop = true;
+                            int errno = Marshal.GetLastWin32Error();
+                            if (errno == EINTR)
+                            {
+                                loop = true;
+                            }
+                            else if (errno == EAGAIN)
+                            {
+                                result = LogResult.Busy;
+                            }
+                            else
+                            {
+                                result = LogResult.UnknownError;
+                                ErrorWhileLogging($"errno={errno}");
+                            }
                         }
-                        else if (errno == EAGAIN)
-                        {
-                            result = LogResult.Busy;
-                        }
-                        else
-                        {
-                            result = LogResult.UnknownError;
-                            ErrorWhileLogging($"errno={errno}");
-                        }
-                    }
-                } while (loop);
+                    } while (loop);
+                }
+                return result;
             }
-            for (int i = 0; i < handles.Length; i++)
+            finally
             {
-                handles[i].Free();
+                for (int i = 0; i < handlesAllocated; i++)
```

Comment:
@jkotas this implements the changes you suggested in https://github.com/tmds/Tmds.Systemd/issues/35. I had assumed this code wouldn't throw. Do you see some exception that may happen here?
I'm fine adopting the change for best practices reasons.

Reply:
`GCHandle.Alloc` can throw OutOfMemoryException.